addons/api/addon/handlers/indexed-db-handler.js

```diff
@@ -0,0 +1,119 @@
+import { modelIndexes } from 'api/services/indexed-db';
+import { inject as service } from '@ember/service';
+import { getOwner, setOwner } from '@ember/application';
+import { pluralize } from 'ember-inflector';
+
+/**
+ * Handler to sit in front of the API request layer
+ * so we can handle any caching with indexedDb first
+ */
+export default class IndexedDbHandler {
+  @service indexedDb;
+
+  constructor(context) {
+    setOwner(this, getOwner(context));
+  }
+
+  async request(context, next) {
```

Comment:
Didn't we already add an indexed-db-handler in search and filtering LLB?

Reply:
Yeah, this is based off of that. I started with that but heavily modified it